scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobState.java

```diff
@@ -193,6 +194,104 @@ public JobPriority getPriority() {
      */
     public abstract Map<TaskId, TaskState> getHMTasks();
 
+
+    /**
+     * To get the task as an array list and filtered by a given tag.
+     * @param tag the used to filter the tasks.
+     * @return the set of filtered task states.
+     */
+    public List<TaskState> getTaskByTag(final String tag){
+        List<TaskState> tasks = this.getTasks();
+        return (List<TaskState>) CollectionUtils.select(tasks, new Predicate() {
+            @Override
+            public boolean evaluate(Object object) {
+                String taskTag = ((TaskState) object).getTag();
+                return (taskTag != null) && (taskTag.equals(tag));
+            }
+        });
+    }
+    
+    
+    /**
+     * To get the paginated tasks.
+     * @param offset the starting index of the sublist of tasks to get
+     * @param limit the last index (non inclusive) of the sublist of tasks to get
+     * @return the list of tasks paginated with [offset, limit[
+     */
+    public List<TaskState> getTasksPaginated(final int offset, final int limit) {
+     // check whether the indexes are inside the list boundaries or not
+        // if they're not, translate indexes inside the boundaries
+        int _offset = 0;
+        int _limit = getTasks().size();
```

Comment:
Would be nice to factorize offset and limit computation since it is used also by the next method.